core/trino-main/src/main/java/io/trino/util/FastutilSetHelper.java

```diff
@@ -318,4 +424,48 @@ public synchronized T get()
             }
         }
     }
+
+    private static Set<Long> createSetForLongJavaType(Set<Long> values, Type type, MethodHandle hashCodeHandle, MethodHandle equalsHandle)
+    {
+        if (values.isEmpty() || !canSkipTypeSpecificEqualsAndHashCode(type)) {
+            return new LongOpenCustomHashSet(values, 0.25f, new LongStrategy(hashCodeHandle, equalsHandle, type));
+        }
+
+        Iterator<Long> iterator = values.iterator();
```

Comment:
do we need iterator here? can we just do a for loop?

this could be just:

```
 long min = Long.MAX_VALUE;
        long max = Long.MIN_VALUE;
        for (Long value : values) {
            min = min(min, value);
            max = max(max, value);
        }
```

Reply:
that works too